datafusion/physical-expr/src/expressions/cast.rs

```diff
@@ -170,7 +170,8 @@ impl PhysicalExpr for CastExpr {
         let target_type = &self.cast_type;
 
         let unbounded = Interval::make_unbounded(target_type)?;
-        if source_datatype.is_numeric() && target_type.is_numeric()
+        if (source_datatype.is_numeric() || source_datatype == Boolean)
```

Comment:
I don't see any test coverage for this change

Reply:
@berkaysynnada let's add a test for this if there is none